floor_generator/lib/processor/queryable_processor.dart

```diff
@@ -21,33 +23,81 @@ abstract class QueryableProcessor<T extends Queryable> extends Processor<T> {
       : assert(classElement != null),
         _queryableProcessorError = QueryableProcessorError(classElement);
 
+  List<FieldElement> get _allFields => [
+        ...classElement.fields,
+        ...classElement.allSupertypes.expand((type) => type.element.fields),
+      ];
+
   @nonNull
   @protected
   List<Field> getFields() {
     if (classElement.mixins.isNotEmpty) {
       throw _queryableProcessorError.prohibitedMixinUsage;
     }
-    final fields = [
-      ...classElement.fields,
-      ...classElement.allSupertypes.expand((type) => type.element.fields),
-    ];
 
-    return fields
+    return _allFields
         .where((fieldElement) => fieldElement.shouldBeIncluded())
         .map((field) => FieldProcessor(field).process())
         .toList();
   }
 
   @nonNull
   @protected
-  String getConstructor(final List<Field> fields) {
+  List<Embedded> getEmbeddeds() {
+    return _allFields
+        .where((fieldElement) => fieldElement.shouldEmbedded())
+        .map((embedded) => EmbeddedProcessor(embedded).process())
+        .toList();
+  }
+
+  @nonNull
+  @protected
+  String getConstructor(
+    final List<Field> fields,
+    final List<Embedded> embeddeds,
+  ) {
+    final constructorBuffer = StringBuffer();
     final constructorParameters = classElement.constructors.first.parameters;
-    final parameterValues = constructorParameters
-        .map((parameterElement) => _getParameterValue(parameterElement, fields))
-        .where((parameterValue) => parameterValue != null)
-        .join(', ');
 
-    return '${classElement.displayName}($parameterValues)';
+    constructorBuffer.write('${classElement.displayName}(');
+
+    constructorBuffer.write(
+      constructorParameters
+          .map((parameterElement) =>
+              _getParameterValue(parameterElement, fields))
+          .where((parameterValue) => parameterValue != null)
+          .join(', '),
+    );
+
+    for (final embedded in embeddeds) {
+      final parameterName = embedded.fieldElement.displayName;
+      final field = constructorParameters.firstWhere(
+        (parameterElement) => parameterElement.name == parameterName,
+        orElse: () => null,
+      );
+
+      constructorBuffer.write(', ');
+
+      if (field.isNamed) {
+        constructorBuffer.write('$parameterName: ');
+      }
+
+      constructorBuffer.write('${embedded.classElement.displayName}(');
+
+      constructorBuffer.write(
+        embedded.classElement.constructors.first.parameters
+            .map((parameterElement) =>
+                _getParameterValue(parameterElement, embedded.fields))
+            .where((parameterValue) => parameterValue != null)
+            .join(', '),
+      );
+
+      constructorBuffer.write(')');
+    }
```

Comment:
❓Does this mean that embedded fields can only be the last constructor parameters?
The following wouldn't work, right?
```dart
class Person {
  @primaryKey
  final int id;

  @embedded
  final Address address;

  final String name;

  Person(this.id, this.address, this.name);
}

class Address {
  final String street;

  Address(this.street);
}
```